src/app/loadout/plug-drawer/getMaximumPotentiallyAvailableArmorSlotEnergyCapacities.ts

```diff
@@ -0,0 +1,138 @@
+    [PlugCategoryHashes.EnhancementsV2Legs]: 0,
+    [PlugCategoryHashes.EnhancementsV2ClassItem]: 0,
+  });
+
+// Thanks ChatGPT
+function getUniquePermutations(nums: number[]): number[][] {
+  const result: number[][] = [];
+
+  function backtrack(start: number) {
+    if (start === nums.length) {
+      result.push([...nums]);
+      return;
+    }
+
+    const used: Set<number> = new Set<number>();
+
+    for (let i = start; i < nums.length; i++) {
+      if (used.has(nums[i])) {
+        continue;
+      }
+
+      [nums[start], nums[i]] = [nums[i], nums[start]];
+      backtrack(start + 1);
+      [nums[start], nums[i]] = [nums[i], nums[start]];
+
+      used.add(nums[i]);
+    }
+  }
+
+  backtrack(0);
+  return result;
+}
+type CostList = [number, number, number, number, number];
+export const sumCostLists = (costLists: CostList[]): CostList => {
+  const res: CostList = [0, 0, 0, 0, 0];
+  for (const costList of costLists) {
+    for (let i = 0; i < costList.length; i++) {
+      res[i] += costList[i];
+    }
+  }
+  return res;
+};
+
+// Pad an array to length 5 with 0s
+export const padCostArray = (arr: number[]) =>
+  Array.from({ ...arr, length: 5 }, (v) => v ?? 0) as CostList;
+
+export function getMaximumPotentiallyAvailableArmorSlotEnergyCapacities(
+  allSelectedMods: PluggableInventoryItemDefinition[]
+): SlotSpecificPlugCategoryHashToNumberRecord {
+  const maximumPotentiallyAvailableArmorSlotEnergyCapacities =
+    getDefaultSlotSpecificPlugCategoryHashToNumberRecord();
+  const selectedSlotSpecificCosts = getDefaultSlotSpecificPlugCategoryHashToNumberRecord();
+  const generalModCosts: number[] = [];
+  const specialtySocketModCosts: number[] = [];
+  for (const selectedMod of allSelectedMods) {
+    const { plugCategoryHash } = selectedMod.plug;
+    if (plugCategoryHash && plugCategoryHash in selectedSlotSpecificCosts) {
+      selectedSlotSpecificCosts[plugCategoryHash as SlotSpecificPlugCategoryHashesKey] +=
+        selectedMod.plug.energyCost?.energyCost || 0;
+    }
+    // Collect general mod costs (Stat boosting mods)
+    if (selectedMod.plug.plugCategoryHash === PlugCategoryHashes.EnhancementsV2General) {
+      generalModCosts.push(selectedMod.plug.energyCost?.energyCost || 0);
+    }
+    // Collect specialty socket mod costs (Raid/nightmare mods)
+    // Exclude artifice mods. They have no cost, and their exclusivity with
+    // raid/nightmare mods is handled elsewhere.
+    const specialtyModslotSelectedMod = [selectedMod]
+      .filter(
+        (selectedMod) =>
+          selectedMod.plug.plugCategoryHash !== PlugCategoryHashes.EnhancementsArtifice
+      )
+      .find((selectedMod) =>
+        Boolean(
+          modTypeTagByPlugCategoryHash[selectedMod.plug.plugCategoryHash as PlugCategoryHashes]
+        )
+      );
+
+    if (specialtyModslotSelectedMod) {
+      specialtySocketModCosts.push(specialtyModslotSelectedMod.plug.energyCost?.energyCost || 0);
+    }
+  }
```

Comment:
We have `categorizeArmorMods` that does something similar (currently it requires a list of reference items with sockets so that we don't end up including deprecated mods from old loadouts but we could make that optional since we know here that the mods are valid.